core/src/test/java/org/apache/calcite/rex/RexExecutorTest.java

```diff
@@ -175,6 +177,58 @@ private void checkConstant(final Object operand,
     });
   }
 
+  @Test public void testUserFromContext() throws Exception {
+    testContextLiteral(SqlStdOperatorTable.USER,
+                       DataContext.Variable.USER, "happyCalciteUser");
+  }
+
+  @Test public void testSystemUserFromContext() throws Exception {
+    testContextLiteral(SqlStdOperatorTable.SYSTEM_USER,
+                       DataContext.Variable.SYSTEM_USER, "");
+  }
+
+  @Test public void testTimestampFromContext() throws Exception {
+    // current timestamp currently rounds to the nearest second (?)
+    long val = System.currentTimeMillis() / 1000 * 1000;
+    testContextLiteral(SqlStdOperatorTable.CURRENT_TIMESTAMP,
+                       DataContext.Variable.CURRENT_TIMESTAMP, val);
+  }
```

Comment:
I think this test will fail intermittently. Rounding to seconds will make it less frequent but doesn't eliminate the problem.

Reply:
How can we address this?

Reply:
Well, if the test starts at 10,000 and finishes at 12,000 then any timestamp value between 10,000 and 12,000 should be considered valid.